x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_alert_types/threshold/lib/api.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { HttpServiceBase } from 'kibana/public';
+
+const WATCHER_API_ROOT = '/api/watcher';
+
+// TODO: replace watcher api with the proper from alerts
+
+export async function getMatchingIndicesForThresholdAlertType({
+  pattern,
+  http,
+}: {
+  pattern: string;
+  http: HttpServiceBase;
+}): Promise<Record<string, any>> {
+  if (!pattern.startsWith('*')) {
+    pattern = `*${pattern}`;
+  }
+  if (!pattern.endsWith('*')) {
+    pattern = `${pattern}*`;
+  }
+  const { indices } = await http.post(`${WATCHER_API_ROOT}/indices`, {
+    body: JSON.stringify({ pattern }),
+  });
+  return indices;
+}
+
+export async function getThresholdAlertTypeFields({
+  indexes,
+  http,
+}: {
+  indexes: string[];
+  http: HttpServiceBase;
+}): Promise<Record<string, any>> {
+  const { fields } = await http.post(`${WATCHER_API_ROOT}/fields`, {
+    body: JSON.stringify({ indexes }),
+  });
+  return fields;
+}
+
+let savedObjectsClient: any;
```

Comment:
Sorry to repeat a point, just don't want this to be missed (it's a big PR, so hard to keep track of it all): we have another module with state, which makes it hard to test and reuse.
React makes it really easy to avoid these types of modules, and we should take advantage of that.